### ISSUE TITLE
Investigate ways to make `multi_gpu_model` faster.

### ISSUE THREAD

**Issue by gabrieldemarmiesse** [keras-team/keras#11775]
Investigate ways to make `multi_gpu_model` faster by trying different aggregation methods (e.g. compute predictions and loss on each device then average gradients across devices, use input pre-fetching, etc).

Note created by @fchollet in the "Requests for contributions".

**Comment by pranavdurai10** [comment]
Hey @gabrieldemarmiesse, is this issue still open for contribution?

**Comment by gabrieldemarmiesse** [comment]
Hi @pranavdurai10  not really, multi-backend keras is deprecated and shouldn't be used anymore. Please use tf.keras instead.
